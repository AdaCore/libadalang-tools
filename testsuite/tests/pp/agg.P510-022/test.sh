echo test number 1
cd 1
gnatpp -pipe -q -P agg.gpr
cd ..

echo test number 2
cd 2
gnatpp -q --pipe -XT=a_val -P agg.gpr
cd ..

echo test number 3
cd 3
gnatpp -q -P agg1.gpr
echo "----------------"
echo src_c/c.ads
cat  src_c/c.ads
echo "----------------"
echo src_c/cc.ads
cat  src_c/cc.ads
echo "----------------"
echo src_d/d.ads
cat  src_d/d.ads
echo "----------------"
echo src_d/dd.ads
cat  src_d/dd.ads
gnatmake -q -P agg1.gpr
cd ..
