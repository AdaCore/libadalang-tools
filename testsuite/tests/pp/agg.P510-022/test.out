test number 1
PACKAGE a1 IS
     I : Integer;
END a1;
test number 2
PACKAGE a IS
     TYPE Int IS RANGE 1 .. 10;
     X : Int;

     Y : Boolean;

     Z : Float;
END a;
test number 3
----------------
src_c/c.ads
PACKAGE C IS
     D : Boolean;
     S : String := "ololo";

     I : Integer;
END C;
----------------
src_c/cc.ads
PACKAGE CC IS
     D : Boolean;
     S : String := "ololo";

     I : Integer;

     A : ARRAY (1 .. 10) OF Integer;
END CC;
----------------
src_d/d.ads
package D is
  I : Integer;
  F : Float;
end D;
----------------
src_d/dd.ads
package DD is
  I : Integer;
  F : Float;

  A : array (1 .. 10) of Integer;
end DD;
